### PR TITLE
added RefitRestServiceAttribute for HttpClient relative path

### PR DESCRIPTION
**Pull request by yuri-voloshyn**
added RefitRestServiceAttribute for HttpClient relative path
fixed issue #662

### CODE REVIEW

File: Refit.Tests/RestService.cs [hunk 1]
```diff
@@ -139,6 +139,24 @@ public interface ITrimTrailingForwardSlashApi
         Task Get();
     }
 
+    [RefitRestService("/api/Test")]
```
Comment (onovotny) [inline Refit.Tests/RestService.cs]: Seems like this should be called `BaseAddressAttribute`?

Comment (yuri-voloshyn) [reply Refit.Tests/RestService.cs]: I renamed attribute

**Comment by pit-hub** [comment]
Hi! I've made some improvements to the work of @yuri-voloshyn.
The idea is not to modify the BaseAddress of the HttpClient. Modifying the BaseAddress may leads to unexpected behaviour by developers and also prevent reusing a HttpClient for multiple interfaces, as each instance of HttpClient uses its own pool of sockets.
Can push to the @yuri-voloshyn fork to update the PR.
The code is at: https://github.com/pit-hub/refit/tree/feature/refit-rest-service-attribute

**Comment by onovotny** [comment]
Sorry, merged this by mistake. @pit-hub can you please open a PR with the combined changes?

**Pull request merged**